crates/payload/basic/src/lib.rs

```diff
@@ -765,6 +782,32 @@ where
     // create the block header
     let transactions_root = proofs::calculate_transaction_root(&executed_txs);
 
+    // initialize empty blob sidecars at first. If cancun is active then this will
+    let mut blob_sidecars = Vec::new();
+    let mut excess_blob_gas = None;
+    let mut blob_gas_used = None;
+
+    // only determine cancun fields when active
+    if chain_spec.is_cancun_activated_at_timestamp(attributes.timestamp) {
+        // grab the blob sidecars from the executed txs
+        let blobs = pool.get_all_blobs(executed_txs.iter().map(|tx| tx.hash).collect())?;
```

Comment:
this also includes non eip4844 transactions?

should filter before (is_eip4844), otherwise we wast blobstore lookups

Reply:
makes sense, added